### PR TITLE
Tril Tests for ladd / lsub

### PR DESCRIPTION
**Pull request by aarongraham9**
Tril Tests for ladd / lsub

For #3105.

Signed-off-by: Aaron Graham <aaron.graham@unb.ca>

### CODE REVIEW

File: fvtest/tril/test/CompileTest.cpp [hunk 1]
```diff
@@ -56,3 +56,37 @@ TEST_F(CompileTest, NoCodeGen) {
...
+TEST_F(CompileTest, Ladd) {
+    auto trees = parseString("(method name=\"ladd\" return=\"Int64\" args=[\"Int64\", \"Int64\"] (block (ireturn (ladd (lload parm=0) (lload parm=1)))))");
+
+    ASSERT_NOTNULL(trees);
+
+    Tril::DefaultCompiler compiler{trees};
+
+    ASSERT_EQ(0, compiler.compile()) << "Compilation failed";
+
+    auto entry = compiler.getEntryPoint<int32_t (*)(int)>();
```
Comment (knn-k) [inline fvtest/tril/test/CompileTest.cpp]: `getEntryPoint<int32_t (*)(int)>()` should be `getEntryPoint<int64_t (*)(int64_t, int64_t)>()` when the compiled method takes two `Int64` arguments and returns an `Int64`.

File: fvtest/tril/test/CompileTest.cpp [hunk 2]
```diff
@@ -56,3 +56,37 @@ TEST_F(CompileTest, NoCodeGen) {
 
     EXPECT_TRUE(verifier.hasRun());
 }
+
+TEST_F(CompileTest, Ladd) {
+    auto trees = parseString("(method name=\"ladd\" return=\"Int64\" args=[\"Int64\", \"Int64\"] (block (ireturn (ladd (lload parm=0) (lload parm=1)))))");
```
Comment (knn-k) [inline fvtest/tril/test/CompileTest.cpp]: `ireturn` should be `lreturn` when the method has the return type `Int64`.

File: fvtest/tril/test/CompileTest.cpp [hunk 3]
```diff
@@ -56,3 +56,37 @@ TEST_F(CompileTest, NoCodeGen) {
...
+
+    ASSERT_NOTNULL(trees);
+
+    Tril::DefaultCompiler compiler{trees};
+
+    ASSERT_EQ(0, compiler.compile()) << "Compilation failed";
+
+    auto entry = compiler.getEntryPoint<int32_t (*)(int)>();
+    ASSERT_NOTNULL(entry) << "Entry point of compiled body cannot be null";
+    ASSERT_EQ(0, entry(0)) << "Compiled body did not return expected value";
```
Comment (knn-k) [inline fvtest/tril/test/CompileTest.cpp]: You need to pass two arguments to `entry()` in this case.

**Comment by aarongraham9** [comment]
I've addressed your comments @knn-k and @Leonardo2718; would you be able to take another look?

File: fvtest/compilertriltest/ArithmeticTest.cpp [hunk 4]
```diff
@@ -80,6 +90,41 @@ TEST_P(Int32Arithmetic, UsingLoadParam) {
     ASSERT_EQ(param.oracle(param.lhs, param.rhs), entry_point(param.lhs, param.rhs));
 }
 
+TEST_P(Int64Arithmetic, UsingConst) {
+    auto param = TRTest::to_struct(GetParam());
+
+    char inputTrees[120] = {0};
+    std::snprintf(inputTrees, 120, "(method return=Int64 (block (lreturn (%s (lconst %d) (lconst %d)) )))", param.opcode.c_str(), param.lhs, param.rhs);
```
Comment (Leonardo2718) [inline fvtest/compilertriltest/ArithmeticTest.cpp]: The format specifiers for the constants should be `%lld`.

**Comment by Leonardo2718** [comment]
Why is "Evaluators" in parentheses? Come to think of, it's not really needed as the tests look more at opcode behaviour rather than the evaluators themselves.

**Comment by Leonardo2718** [comment]
@genie-omr build all

**Pull request merged**